contracts/utilities/IUtilityChecker.sol

```diff
@@ -0,0 +1,123 @@
+//                                        :@@@#%@%=
+//                                        +@@@@%-
+//                                        :#%%=
+//
+
+/**
+ *     NOTICE
+ *
+ *     The T-REX software is licensed under a proprietary license or the GPL v.3.
+ *     If you choose to receive it under the GPL v.3 license, the following applies:
+ *     T-REX is a suite of smart contracts implementing the ERC-3643 standard and
+ *     developed by Tokeny to manage and transfer financial assets on EVM blockchains
+ *
+ *     Copyright (C) 2023, Tokeny sàrl.
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.26;
+
+import { IClaimIssuer } from "@onchain-id/solidity/contracts/interface/IClaimIssuer.sol";
+
+
+interface IUtilityChecker {
+
+    struct ComplianceCheckDetails {
+        string moduleName;
+        bool pass;
+    }
+
+    struct EligibilityCheckDetails {
+        IClaimIssuer issuer;
+        uint256 topic;
+        bool pass;
+    }
+
+    /// @dev This function verifies if the transfer is restricted due to frozen addresses or tokens.
+    /// @param _token The address of the token contract.
+    /// @param _from The address of the sender.
+    /// @param _to The address of the recipient.
+    /// @param _amount The amount of tokens to be transferred.
+    /// @return bool Returns true if the transfer is not affected by freeze conditions, false otherwise.
+    /// @return uint256 Available unfreezed balance.
+    function testFreeze(address _token, address _from, address _to, uint256 _amount) external view returns (bool, uint256);
+
+    /// @dev This function performs a comprehensive check on whether a transfer would succeed:
+    ///     - check if token is paused,
+    ///     - check freeze conditions,
+    ///     - check compliance.
+    /// @param _compliance Address of the Modular Compliance contract.
+    /// @param _token The address of the token contract.
+    /// @param _from The address of the sender.
+    /// @param _to The address of the recipient.
+    /// @param _amount The amount of tokens to be transferred.
+    /// @return bool Returns true if the transfer would be successful, false otherwise.
+    function testTransfer(address _compliance, address _token, address _from, address _to, uint256 _amount) 
```

Comment:
the interface can be simplified by removing the compliance address, this address can be found easily if you have the token address as there is a getter on the token interface to get the compliance address.